### PR TITLE
Terraform New Relic Slack Workflow for Staging Notifications

### PR DESCRIPTION
**Pull request by P0NDER0SA**
# Summary | Résumé

Adding a codified TF solution for our Slack Staging notifications

## Related Issues | Cartes liées

https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/381

# Test instructions | Instructions pour tester la modification

1. ensure terragrunt plan works
2. navigate to this workflow and get it to send a test notification to slack after terragrunt apply

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: .github/workflows/merge_to_main_production.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ env:
   AWS_REGION: ca-central-1
   TF_VAR_new_relic_api_key: ${{ secrets.PRODUCTION_NEW_RELIC_API_KEY }}
   TF_VAR_new_relic_account_id: ${{ secrets.PRODUCTION_NEW_RELIC_ACCOUNT_ID }}
+  TF_VAR_new_relic_new_relic_slack_webhook_url: ${{ secrets.PRODUCTION_NEW_RELIC_SLACK_WEBHOOK_URL }}
```
Comment (ben851) [inline .github/workflows/merge_to_main_production.yml:22]: new_relic_new_relic?

Comment (P0NDER0SA) [reply .github/workflows/merge_to_main_production.yml]: fixed that.  thanks

File: aws/newrelic/alert_policies.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
-# resource "newrelic_alert_policy" "notify_terraform_policy" {
-#   name     = "Pond Example staging"
-#   provider = newrelic
-# }
+resource "newrelic_alert_policy" "terraform_notify_policy" {
+  name     = "TF Notify Policy - ${var.env}"
```
Comment (jimleroyer) [inline aws/newrelic/alert_policies.tf:2]: The `TF` prefix is temporary just to work with it in the meantime along the existing policies?

Comment (P0NDER0SA) [reply aws/newrelic/alert_policies.tf:2]: That's right @jimleroyer --  just so we can easily identify the TF ones in the meantime.

File: env/production/newrelic/terragrunt.hcl [hunk 3]
```diff
@@ -1,5 +1,20 @@
+# Uses GitHub tags for release management
+#
 terraform {
-  source = "../../../aws//newrelic"
+  source = "git::https://github.com/cds-snc/notification-terraform//aws/newrelic?ref=v${get_env("INFRASTRUCTURE_VERSION")}"
```
Comment (ben851) [inline env/production/newrelic/terragrunt.hcl:4]: 👍

File: env/production/newrelic/terragrunt.hcl [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+    execute      = ["git", "clone", "https://github.com/cds-snc/notification-admin.git", "/var/tmp/notification-admin"]
+    run_on_error = true
+
+  }
+
+  after_hook "cleanup-admin" {
+    commands     = ["apply", "plan"]
+    execute      = ["rm", "-rfd", "/var/tmp/notification-admin"]
+    run_on_error = true
+  }
```
Comment (ben851) [inline env/production/newrelic/terragrunt.hcl:17]: This can be removed

**Review by ben851 — APPROVED** [review]

File: env/production/newrelic/terragrunt.hcl (hunk 4 above)
Comment (P0NDER0SA) [reply env/production/newrelic/terragrunt.hcl]: removed

**Pull request merged**